lib/util/Components.js

```diff
@@ -798,6 +800,17 @@ function componentRule(rule, context) {
 
       // Return the component
       return components.add(componentNode, 1);
+    },
+
+    isArrowOrFunctionExpressionRenderWithParams(node) {
+      return (
+        node.parent
+        && node.parent.key
+        && node.parent.key.type === 'Identifier'
+        && node.parent.key.name === 'render'
+        // react render function cannot have params
```

Comment:
```suggestion
        && node.parent.key.name.charAt(0) === node.parent.key.name.charAt(0).toLowerCase()
        // custom component functions must start with a capital letter
```
what about something like this? there's nothing inherentky magical about the word "render".

Reply:
Yeah, it makes sense, thanks

Reply:
Now that I test it - it breaks tests like these

no-unstable-nested-components

```
      function ParentComponent() {
        const rows = [
          {
            name: 'A',
            notPrefixedWithRender: (props) => <Row {...props} />
          },
        ];

        return <Table rows={rows} />;
      }
```
This should return an error since `notPrefixedWithRender:` is an unstable nested component but does not since according to this new logic proposed here  `notPrefixedWithRender:` is not a component. 

This test can be re-written but IMO a lot of external libraries like ant-d, mui have API like above, so it can introduce breaking changes.

```
<some-lowercase-firstletter>Anything: () => jsx
```

Reply:
That’s not a component at all - it’s a render prop. Components must start with a capital letter.

Reply:
Specifically, in react, anything that starts with a lowercase letter is an html element, not a component.